### DISCUSSION TITLE
Detecting if navigated to using the back button/replace history

### DISCUSSION THREAD

**Discussion by nickmask** [vercel/next.js#12556] [Help]
I've got a situation where the user is stuck in a bit of a redirect loop it they navigate back to a page using the browser's back button.

I am trying to find the best solution to either remove/modify the previous browser history or detect if the user navigated to a page using back button and handle that differently.

Any ideas?

**Comment by tamlyn** [comment]
React Router has a [`history.action` property](https://reactrouter.com/web/api/history) which indicates whether the last action was a push, replace or pop. But as far as I can tell Next Router doesn't have this or an equivalent.

**Reply by Amp-kebinjoy** [reply]
I want to differentiate the browser forward and back button actions so I can conditionally render in-app buttons for navigation. I could go something like this:
` useEffect(() => {
        Router.router.beforePopState((e) => {
                // do something
        });
}, []);
`
this will be fired for browser button actions. But the issue is that it doesn't differentiate the back and forward buttons. It just gets fired all the time and there is no data in e to figure out which button was pressed.

This is kinda incomplete because if go like page1 => page2 => page1, and press "back" and is on page 2, pressing forward or back will land me on page1 but I need to know if it was done via which button to conditionally render in-app nav buttons.

Any suggestions, please?
